pkg/sql/logictest/testdata/logic_test/role

```diff
@@ -181,18 +181,33 @@ user root
 statement ok
```

Comment:
can we run this logic test file with cluster setting on and off?
add something to https://github.com/cockroachdb/cockroach/blob/c6a5a395cbecbbd5ba77c2c14e3bf68eb59eeb19/pkg/sql/logictest/logic.go#L679-L684 and add `LogicTest: local .... local-cache-single-read-off` or something?

Reply:
added a metamorphic constant